### PR TITLE
Add assertions to IPU accelerator for device availability

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?

Add assertions to IPU accelerator for device availability
This aims to simplify the accelerator connector logic and rewrite effort in https://github.com/PyTorchLightning/pytorch-lightning/pull/11448

This would move the assertion logic from Trainer constructor checks to runtime at trainer.fit/validate/test/predict calls

Related PR: https://github.com/PyTorchLightning/pytorch-lightning/pull/11797
## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by four4fish — APPROVED** [review]

File: pytorch_lightning/accelerators/ipu.py [hunk 1]
```diff
@@ -16,11 +16,25 @@
...
 class IPUAccelerator(Accelerator):
     """Accelerator for IPUs."""
 
+    def setup_environment(self, root_device: torch.device) -> None:
+        """
+        Raises:
+            MisconfigurationException:
+                If the IPU device is not available.
+        """
+        if not _IPU_AVAILABLE:
```
Comment (carmocca) [inline pytorch_lightning/accelerators/ipu.py:32]: Move to __init__?

**Review by carmocca — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -98,6 +98,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added a `_Stateful` support for `LightningDataModule` ([#11637](https://github.com/PyTorchLightning/pytorch-lightning/pull/11637))
 
 
+- Added checks to `IPUAccelerator` to assert device availability ([#11799](https://github.com/PyTorchLightning/pytorch-lightning/pull/11798))
```
Comment (rohitgr7) [inline CHANGELOG.md:101]:
```suggestion
- Added checks to `IPUAccelerator` to assert device availability ([#11798](https://github.com/PyTorchLightning/pytorch-lightning/pull/11798))
```

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/accelerators/ipu.py [hunk 3]
```diff
@@ -16,11 +16,25 @@
...
 
+    def __init__(self) -> None:
+        """
+        Raises:
+            MisconfigurationException:
+                If the IPU device is not available.
+        """
+        if not _IPU_AVAILABLE:
+            raise MisconfigurationException(
+                "The IPU Accelerator requires IPU devices to run. "
```
Comment (tchaton) [inline pytorch_lightning/accelerators/ipu.py:34]:
```suggestion
                "The IPU Accelerator requires `GraphCore IPU devices` to run. "
```

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by ananthsub** [comment]
Closing out in favor of the approach in https://github.com/PyTorchLightning/pytorch-lightning/pull/11797

**Pull request closed without merging**